### DISCUSSION TITLE
WeMo emulation doesn't always work

### DISCUSSION THREAD

**Discussion by bmustill-rose** [arendst/Tasmota#18384] [Support request]
Hi all,

I've installed a smart plug for a family member with WeMo emulation enabled. They typically trigger it through an Echo routine. At some point in the past month something's changed though and the plug now doesn't always respond through the routine and sometimes when triggered manually through the Echo - E.G. turn on xxx. It's currently running 12.4 but the issue was also present under 12.3.1. I haven't tested any other versions.

I initially thought that it was a connectivity issue, but I note that log entries are showing up when the Echo runs the routine:

19:32:12.523 WMO: WeMo Type 2, Response sent to 192.168.0.190:50000
19:32:12.573 HTP: WeMo setup from 192.168.0.190
19:32:12.678 HTP: WeMo basic event from 192.168.0.190
19:32:15.630 WMO: WeMo Type 2, Response sent to 192.168.0.190:50000
19:32:18.636 WMO: WeMo Type 2, Response sent to 192.168.0.190:50000
19:33:12.476 HTP: WeMo basic event from 192.168.0.190

Those are all the entries for one trigger of the routine. 0.190 is a fourth gen Echo. Note that the plug consistently works when I control it through the HTTP API.

Any ideas will be very much appreciated.
Thanks!

**Comment by pkkrusty** [comment]
I have always had more consistent function when using Hue emulation instead.
